src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn3.11.cs

```diff
@@ -117,26 +140,17 @@ public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
         /// <summary>
         /// Instrumentation helper for unit tests.
         /// </summary>
-        public Action<SourceGenerationSpec?>? OnSourceEmitting { get; init; }
-    }
-
-    internal readonly struct JsonSourceGenerationContext
-    {
-        private readonly GeneratorExecutionContext _context;
+        public Action<ImmutableArray<ContextGenerationSpec>>? OnSourceEmitting { get; init; }
 
-        public JsonSourceGenerationContext(GeneratorExecutionContext context)
+        private partial class Emitter
         {
-            _context = context;
-        }
+            private readonly GeneratorExecutionContext _context;
 
-        public void ReportDiagnostic(Diagnostic diagnostic)
-        {
-            _context.ReportDiagnostic(diagnostic);
-        }
+            public Emitter(in GeneratorExecutionContext context)
```

Comment:
Why `in`?

Reply:
Seems to have come from here:

https://github.com/dotnet/runtime/pull/86616/files#diff-b7da6b90947be2bdebc35f367d9dc2c5bb66dd3be881af8a2d6cbb9c18108a7fL86